src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -398,15 +398,19 @@ u16 EnGo2_GetTextIdGoronDmtDcEntrance(PlayState* play, EnGo2* this) {
         return 0x3043;
     } else if (CHECK_QUEST_ITEM(QUEST_GORON_RUBY)) {
         return 0x3027;
+    } else if (GET_EVENTCHKINF(EVENTCHKINF_DESTROYED_DODONGOS_CAVERN_BOULDER)) {
+        return 0x3021;
+    } else if (GET_INFTABLE(INFTABLE_TALKED_TO_DODONGOS_CAVERN_ENTRANCE_GORON)) {
+        return 0x302A;
     } else {
-        return GET_EVENTCHKINF(EVENTCHKINF_23) ? 0x3021 : GET_INFTABLE(INFTABLE_E0) ? 0x302A : 0x3008;
+        return 0x3008;
```

Comment:
this is at least one thing that looks incorrect, you changed the logic of the function

Reply:
If you follow it, no I didn't. These are conditional statements that I just merged with the if-else conditionals. Functionally, nothing changed.

If you're saying that's not allowed, I'll say that's a load of BS.

Reply:
Do you understand how the matching component of decompilation works?
While the code you wrote may be functionally equivalent, its different enough to result in different assembly instructions after it compiles. The code is written the way it is currently because it matches how they wrote it originally.

Reply:
I want to make sure you understand why what you did doesnt work. The checksum mismatch is because the code change was significant enough to cause a difference in the compiled code. Something like a ternary will result in different code being generated then if/elseifs.

Reply:
I understand that different code compiles different, but that's so alike that there shouldn't have been a difference IMO. It's just two different syntaxes for the exact same thing.

Reply:
You can take it up with the authors of the IDO compiler from Silicon Graphics in the late 90s 😂 
Really though, they do compile differently. There are plenty of things that you may intuitively think will compile to the same assembly instructions, but they dont. This compiler in particular is pretty weird when it comes to certain things.

We spent years matching this game working around all these strange quirks!

Reply:
Perhaps while I'm at it, I'll ask Nintendo why these are ternaries in the first place!

Alright, let's move on from this.

Reply:
@JordanLongstaff, always run a diff when you change logic, don't rely on you opinion
```
.venv/bin/python3 tools/asm-differ/diff.py -mwo3 EnGo2_GetTextIdGoronDmtDcEntrance
# in this case; or use other flags and arguments where applicable
```
also at the moment there's a PR with general logic/data documentation: https://github.com/zeldaret/oot/pull/2295